### PR TITLE
passed the OnnxRuntimeBuildDirectory to the docker for the dotnet linux test

### PR DESCRIPTION
**Pull request by shahasad**

### CODE REVIEW

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh [hunk 1]
```diff
@@ -29,6 +29,7 @@ docker run -h $HOSTNAME \
         --volume "$BUILD_DIR:/home/onnxruntimedev" \
         --volume "$HOME/.cache/onnxruntime:/home/onnxruntimedev/.cache/onnxruntime" \
         "onnxruntime-$IMAGE" \
+        -e "OnnxRuntimeBuildDirectory=$OnnxRuntimeBuildDirectory" \
         /bin/bash /onnxruntime_src/csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh \
         /home/onnxruntimedev/$NUGET_REPO_DIRNAME /onnxruntime_src /home/onnxruntimedev $TestDataUrl $TestDataChecksum &
```
Comment (jignparm) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh]: onnxruntime_src is passed on command line, whereas OnnxRuntimeBuildDirectory is passed as environment variable.

It's a bit confusing for local users. Can we choose 1 pattern (either all environment or all command line?)

Comment (shahasad) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh]: there is difference. The OnnxRuntimeBuildDirectory ultimately need to set as env var for the project file, while the other variables are only used in the script that docker runs.

Comment (jignparm) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh]: Let's add a comment about which ones need to be set by the enviroment (at top of script)

Comment (shahasad) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh]: updated.

**Review by wcy123 — APPROVED** [review]

**Pull request merged**